src/util/specific.py

```diff
@@ -137,3 +138,66 @@ def reset_cowed_views(cr, xmlid, key=None):
         [key, module, name],
     )
     return set(sum(cr.fetchall(), ()))
+
+
+def translation2jsonb_unofficial_fields(cr):
+    """Convert all translated unofficial fields to jsonb.
+
+    This helper function is designed to run only in an on-premise Odoo instance
+    where any unofficial modules you use are installed, after going through the
+    upgrade to version 16.0.
+
+    It will find all those fields and convert translations from the
+    ``_ir_translation`` table into the new JSONB format.
+
+    Any fields not loaded will be skipped.
+    """
+    cr.execute(
+        """
+        SELECT ARRAY_AGG(name)
+        FROM ir_module_module
+        WHERE author LIKE 'Odoo S.A.'
+        """
+    )
+    official_modules = set(cr.fetchone()[0])
+    cr.execute(
+        """
+        SELECT ARRAY_AGG(id)
+        FROM ir_model_fields
+        WHERE store = TRUE AND translate = TRUE
+        """
+    )
+    env = env(cr)
+    all_migrate_queries, all_cleanup_queries = [], []
+    for ir_field in iter_browse(env["ir.model.fields"], cr.fetchall()[0][0]):
+        # Skip official fields, which were translated already
+        if ir_field.modules and official_modules.intersection(ir_field.modules.split(", ")):
+            continue
+        try:
+            model = env[ir_field.model_id.model]
+            field = model._fields[ir_field.name]
+        except KeyError:
+            _logger.warning(
+                "Field %s in model %s is not loaded; skipping",
+                ir_field.name,
+                ir_field.model_id.model,
+            )
+            continue
```

Comment:
This suggestion is very close to how we do in our upgrade script. With possible alternatives to narrow down what to update.
```suggestion
    for Model in env.registry.values():
        if Model._auto and not Model._abstract: # perhaps: and Model._name in models:
            for field in Model._fields.values():
                if field.store and field.translate and not field.manual:                    # perhaps: and (Model._name, field.name) in fields
                   
```